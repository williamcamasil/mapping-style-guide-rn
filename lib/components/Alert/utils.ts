import { ViewStyle } from 'react-native';

import { Icons } from '../../assets';
import { AppThemeType } from '../../theme/default';
import { ColorPaletesType } from '../../tokens';

export type AlertVariantType =
  'success' |
  'info' |
  'warning' |
  'error';

export const getVariantColorStyle = (variant?: AlertVariantType): ColorPaletesType => {
  switch (variant) {
    case 'warning':
      return 'feedbackAlert500';
    case 'error':
      return 'feedbackError500';
    case 'info':
      return 'neutralGray600';
    case 'success':
    default:
      return 'feedbackSuccess500';
  }
};

export const getDefaultIcon = (variant?: AlertVariantType) => {
  switch (variant) {
    case 'success':
      return Icons.Default.Success;
    case 'info':
      return Icons.Default.Information;
    case 'warning':
    case 'error':
    default:
      return Icons.Default.Warning;
  }
};

export const getAlertStyle = (
  theme: AppThemeType,
  variant?: AlertVariantType,
): ViewStyle => {

  switch (variant) {
    case 'warning':
      return {
        backgroundColor: theme.colors.feedbackAlert100,
      };
    case 'error':
      return {
        backgroundColor: theme.colors.feedbackError100,
      };
    case 'info':
      return {
        backgroundColor: theme.colors.neutralGray100,
      };
    case 'success':
    default:
      return {
        backgroundColor: theme.colors.feedbackSuccess100,
      };
  }
};
