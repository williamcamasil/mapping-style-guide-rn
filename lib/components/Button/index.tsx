import React, { ComponentType } from 'react';
import {
  StyleProp, StyleSheet, TouchableOpacity, TouchableWithoutFeedbackProps, View, ViewStyle,
} from 'react-native';

import { SvgProps } from 'react-native-svg';

import { useViewStyles } from '../../hooks/useStyles';
import { withTheme } from '../../theme';
import { AppThemeType } from '../../theme/default';
import Loading from '../Loading';
import Spacer from '../Spacer';
import Text, { TextPropTypes } from '../Text';
import {
  ButtonSizeType,
  ButtonVariantType,
  getSizeStyle,
  getTextColor,
  getVariantStyle,
  getStylesToBlock,
} from './utils';

export type ButtonPropsType<IconProps extends SvgProps = SvgProps> = {
  children?: TextPropTypes['children'];
  onPress: TouchableWithoutFeedbackProps['onPress'];
  style?: StyleProp<ViewStyle>;
  disabled?: TouchableWithoutFeedbackProps['disabled'];
  theme: AppThemeType;
  variant?: ButtonVariantType;
  size?: ButtonSizeType;
  loading?: boolean;
  testID?: string;
  Icon?: ComponentType<IconProps>;
  block?: boolean;
  containerStyle?: StyleProp<ViewStyle>;
  textColorVariant?: ButtonVariantType;
};

const styles = StyleSheet.create({
  containerIconAndText: {
    flexDirection: 'row',
    alignItems: 'center',
  },
});

const Button: React.FC<ButtonPropsType> = ({
  children,
  style,
  onPress,
  disabled,
  theme,
  variant,
  size,
  loading,
  testID,
  Icon,
  block,
  containerStyle,
  textColorVariant,
}) => {
  const hasText = Boolean(children);

  const buttonStyles = useViewStyles(
    () => [
      {
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        borderRadius: theme.borders.radius.medium,
      },
      getVariantStyle(theme, variant, disabled),
      getSizeStyle(theme, size, hasText),
      style,
    ],
    [disabled, hasText, size, style, theme, variant],
  );

  const containerStyles = useViewStyles(() => [getStylesToBlock(block), containerStyle], [block, containerStyle]);

  const textColor = getTextColor(textColorVariant ? textColorVariant : variant, disabled);

  const renderText = () => {
    if (!hasText) return null;
    return (
      <Text
        testID="text-with-icon"
        variant={size === 'small' ? 'bodySmall' : 'body'}
        weight="bold"
        lineHeight="medium"
        color={textColor}
      >
        {children}
      </Text>
    );
  };

  const renderContent = () => {
    if (loading) {
      return <Loading color={textColor} />;
    }

    if (Icon) {
      return (
        <View testID="container-icon" style={styles.containerIconAndText}>
          <Icon width={32} height={32} color={theme.colors[textColor]} />
          {children ? <Spacer size={theme.spacings.sXXS} /> : null}
          {renderText()}
        </View>
      );
    }

    return renderText();
  };

  return (
    <View testID="container-button" style={containerStyles}>
      <TouchableOpacity
        testID={testID}
        style={buttonStyles}
        onPress={onPress}
        disabled={disabled || loading}
        activeOpacity={theme.opacities.intense}
      >
        {renderContent()}
      </TouchableOpacity>
    </View>
  );
};

Button.defaultProps = {
  disabled: false,
  variant: 'containedPrimary',
  size: 'medium',
  block: true,
};

export default withTheme(Button);
