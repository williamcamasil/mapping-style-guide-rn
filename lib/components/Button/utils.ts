import { ViewStyle } from 'react-native';

import { AppThemeType } from '../../theme/default';
import { ColorPaletesType } from '../../tokens';

export type ButtonVariantType =
  'containedPrimary' |
  'containedSecondary' |
  'outlinedPrimary' |
  'outlinedSecondary' |
  'text';

export type ButtonSizeType =
  'large' |
  'medium' |
  'small';

export const getVariantStyle = (
  theme: AppThemeType,
  variant?: ButtonVariantType,
  disabled?: boolean | null,
): ViewStyle => {
  switch (variant) {
    case 'containedSecondary':
      return {
        backgroundColor: disabled ? theme.colors.primary100 : theme.colors.primary200,
        borderWidth: theme.borders.width.none,
      };
    case 'outlinedPrimary':
      return {
        backgroundColor: 'transparent',
        borderWidth: theme.borders.width.thin,
        borderColor: disabled ? theme.colors.primary100 : theme.colors.primary600,
      };
    case 'outlinedSecondary':
      return {
        backgroundColor: 'transparent',
        borderWidth: theme.borders.width.thin,
        borderColor: disabled ? theme.colors.neutralGray100 : theme.colors.neutralGray300,
      };
    case 'text':
      return {
        backgroundColor: disabled ? theme.colors.neutralGray100 : 'transparent',
        borderWidth: theme.borders.width.none,
      };
    case 'containedPrimary':
    default:
      return {
        backgroundColor: disabled ? theme.colors.primary100 : theme.colors.primaryMain,
        borderWidth: theme.borders.width.none,
      };
  }
};

export const getTextColor = (
  variant?: ButtonVariantType,
  disabled?: boolean | null,
): ColorPaletesType => {
  switch (variant) {
    case 'containedSecondary':
      return disabled ? 'primary300' : 'primaryMain';
    case 'outlinedPrimary':
      return disabled ? 'primary300' : 'primaryMain';
    case 'outlinedSecondary':
      return disabled ? 'neutralGray200' : 'neutralGray600';
    case 'text':
      return disabled ? 'neutralGray400' : 'neutralGray500';
    case 'containedPrimary':
    default:
      return disabled ? 'primary300' : 'neutralWhite';
  }
};

export const getSizeStyle = (theme: AppThemeType, size?: ButtonSizeType, hasText?: boolean): ViewStyle => {
  switch (size) {
    case 'large':
      return {
        height: 64,
        minWidth: 64,
        paddingHorizontal: hasText ? theme.spacings.sXL : undefined,
      };
    case 'small':
      return {
        height: 30,
        minWidth: 30,
        paddingHorizontal: hasText ? theme.spacings.sXS : undefined,
      };
    case 'medium':
    default:
      return {
        height: 48,
        minWidth: 48,
        paddingHorizontal: hasText ? theme.spacings.sMedium : undefined,
      };
  }
};

export const getStylesToBlock = (block: boolean | undefined): ViewStyle | null => {
  if (block) return null;

  return {
    flexDirection: 'row',
  };
};
