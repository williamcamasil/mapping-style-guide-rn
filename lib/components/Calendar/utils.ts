import { dateToString } from 'mapping-context-rn';
import type { MarkedDates, Theme } from 'react-native-calendars/src/types';

import { AppThemeType } from '../../theme';

type ExtendedTheme = Theme & {
  'stylesheet.calendar.header': object;
  'stylesheet.day.basic': object;
};

/**
 * Define o tipo de data para o formato yyyy-MM-dd
 */
export type StringDateType = `${number}${number}${number}${number}-${number}${number}-${number}${number}`;

export const DATE_FORMAT = 'yyyy-MM-dd';

export const DAY_SIZE = 42;

export const getDefaultValue = (value?: StringDateType | null) => (
  value ?? dateToString(new Date(), DATE_FORMAT) as StringDateType
);

export const getDisabledDates = (disabledDates?: StringDateType[]) => {
  if (!disabledDates?.length) return {};
  const dates: MarkedDates = {};
  disabledDates.forEach((date: StringDateType) => {
    dates[date] = {
      disabled: true,
      disableTouchEvent: true,
    };
  });
  return dates;
};

export const getCommonTheme = (theme: AppThemeType): ExtendedTheme => ({
  selectedDayBackgroundColor: theme.colors.primaryMain,
  todayTextColor: theme.colors.primaryMain,
  dayTextColor: theme.colors.neutralGray600,
  textDayFontFamily: theme.typography.weights.bold.fontFamily,
  textMonthFontFamily: theme.typography.weights.bold.fontFamily,
  textDayHeaderFontFamily: theme.typography.weights.bold.fontFamily,
  textDayFontWeight: theme.typography.weights.bold.fontWeight,
  textMonthFontWeight: theme.typography.weights.bold.fontWeight,
  textDayHeaderFontWeight: theme.typography.weights.bold.fontWeight,
  textDayFontSize: theme.typography.sizes.small,
  textMonthFontSize: theme.typography.sizes.small,
  textDayHeaderFontSize: theme.typography.sizes.small,
  'stylesheet.calendar.header': {
    dayHeader: {
      marginTop: theme.spacings.sXS,
      marginBottom: theme.spacings.sXS,
      textAlign: 'center',
      fontSize: theme.typography.sizes.small,
      color: theme.colors.primaryMain,
      fontFamily: theme.typography.weights.bold.fontFamily,
      fontWeight: theme.typography.weights.bold.fontWeight,
      flex: 1,
    },
    week: {
      flexDirection: 'row',
      justifyContent: 'space-between',
    },
  },
  'stylesheet.day.basic': {
    base: {
      height: DAY_SIZE,
      width: DAY_SIZE,
      borderBottomLeftRadius: theme.borders.radius.XL,
      borderBottomRightRadius: theme.borders.radius.XL,
      borderTopLeftRadius: theme.borders.radius.XL,
      borderTopRightRadius: theme.borders.radius.XL,
      alignItems: 'center',
      justifyContent: 'center',
    },
  },
});
