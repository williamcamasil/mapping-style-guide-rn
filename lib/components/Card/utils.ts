import { ColorValue, ViewStyle } from 'react-native';

import { AppThemeType } from '../../theme/default';

export type CardSizeType =
  'small' |
  'big';

export type CardVariantType =
  'default' |
  'highlighted50' |
  'highlighted100';

export const getCardSizeStyle = (theme: AppThemeType, size?: CardSizeType): ViewStyle => {
  switch (size) {
    case 'small':
      return {
        width: 135,
        height: 120,
        padding: theme.spacings.sXS,
      };
    case 'big':
    default:
      return {
        minHeight: 128,
        padding: theme.spacings.sSmall,
      };
  }
};

export const getCardShadowStyle = (theme: AppThemeType, showShadow?: boolean): ViewStyle | null => {
  if (showShadow) {
    return theme.shadows.level4;
  }

  return {
    borderWidth: theme.borders.width.thin,
    borderColor: theme.colors.neutralGray200,
  };
};

export const getCardVariantStyle = (theme: AppThemeType, cardType?: CardVariantType): ViewStyle | null => {
  switch (cardType) {
    case 'highlighted50':
      return {
        borderWidth: theme.borders.width.thin,
        borderColor: theme.colors.primary700,
      };
    case 'highlighted100':
      return {
        borderWidth: theme.borders.width.thin,
        borderColor: theme.colors.primary600,
        backgroundColor: theme.colors.primaryMain,
      };
    case 'default':
    default:
      return null;
  }
};

export const getCardCustomBorderStyle = (theme: AppThemeType, borderColor?: ColorValue): ViewStyle | null => {
  if (!borderColor) return null;

  return {
    borderWidth: theme.borders.width.thin,
    borderColor,
  };
};

export const getArrowColor = (theme: AppThemeType, variant?: CardVariantType, arrowColor?: ColorValue): ColorValue => {
  if (arrowColor) return arrowColor;

  if (variant === 'highlighted100') return theme.colors.neutralWhite;

  return theme.colors.neutralGray500;
};
