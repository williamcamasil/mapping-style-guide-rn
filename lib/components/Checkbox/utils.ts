import { CheckBoxImgPropsType } from '../../assets/images/CheckboxImg';
import { AppThemeType } from '../../theme/default';
import { ColorPaletesType } from '../../tokens';

export type CheckboxVariantType = 'default' | 'secondary';

export const getTextColor = (error?: boolean, variant?: CheckboxVariantType, disabled?: boolean): ColorPaletesType => {
  if (disabled) {
    if (variant === 'secondary') return 'secondary500';
    return 'neutralGray300';
  }

  if (error) {
    if (variant === 'secondary') return 'feedbackError300';
    return 'feedbackError400';
  }

  if (variant === 'secondary') return 'secondary200';

  return 'neutralGray700';
};

const getBackgroundColorStyle = (theme: AppThemeType, variant?: CheckboxVariantType) => {
  if (variant === 'secondary') return theme.colors.secondary700;

  return theme.colors.neutralGray100;
};

const getCheckedBorderColorStyle = (theme: AppThemeType, variant?: CheckboxVariantType) => {
  if (variant === 'secondary') return theme.colors.secondary600;

  return theme.colors.neutralGray200;
};

const getColorStyle = (theme: AppThemeType, error?: boolean, variant?: CheckboxVariantType) => {
  if (error) {
    if (variant === 'secondary') return theme.colors.feedbackError300;
    return theme.colors.feedbackError400;
  }

  return theme.colors.primaryMain;
};

const getDisabledBorderColorStyle = (theme: AppThemeType, variant?: CheckboxVariantType) => {
  if (variant === 'secondary') return theme.colors.secondary500;

  return theme.colors.neutralGray200;
};

export const getBoxStyle = (
  theme: AppThemeType,
  error?: boolean,
  disabled?: boolean,
  checked?: boolean,
  variant?: CheckboxVariantType,
): CheckBoxImgPropsType => {
  if (disabled) {
    return (
      {
        backgroundColor: 'transparent',
        borderColor: getDisabledBorderColorStyle(theme, variant),
        borderWidth: theme.borders.width.thin,
        checkColor: theme.colors.neutralGray200,
        checked,
      }
    );
  }

  if (checked) {
    return (
      {
        backgroundColor: getColorStyle(theme, error, variant),
        borderColor: getCheckedBorderColorStyle(theme, variant),
        borderWidth: theme.borders.width.thick,
        checkColor: theme.colors.neutralWhite,
        checked,
      }
    );
  }

  return (
    {
      backgroundColor: getBackgroundColorStyle(theme, variant),
      borderColor: getColorStyle(theme, error, variant),
      borderWidth: theme.borders.width.thin,
      checkColor: theme.colors.neutralGray100,
      checked,
    }
  );
};
