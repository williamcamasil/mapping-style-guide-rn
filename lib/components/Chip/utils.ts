import { AppThemeType } from '../../theme/default';

export const getButtonStyles = (selected: boolean, theme: AppThemeType) => {
  if (selected) {
    return {
      backgroundColor: theme.colors.primary100,
      borderColor: theme.colors.primary300,
    };
  }
  return {
    backgroundColor: 'transparent',
    borderColor: theme.colors.neutralGray300,
  };
};
