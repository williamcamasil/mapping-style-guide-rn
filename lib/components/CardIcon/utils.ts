import { ColorValue } from 'react-native';

import { AppThemeType } from '../../theme/default';
import { CardVariantType } from '../Card/utils';

export const getIconColor = (
  theme: AppThemeType,
  customColor?: ColorValue,
  variant?: CardVariantType,
): ColorValue | string => {
  if (customColor) return customColor;

  if (variant === 'highlighted100') {
    return theme.colors.neutralWhite;
  }

  return theme.colors.primaryMain;
};
