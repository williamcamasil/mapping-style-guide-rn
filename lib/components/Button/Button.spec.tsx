import React from 'react';
import { StyleSheet, ViewStyle } from 'react-native';

import { within, render, fireEvent } from '@testing-library/react-native';

import Button from '.';
import { Icons } from '../../assets';
import { DefaultTheme } from '../../theme';
import Loading from '../Loading';
import Text from '../Text';
import { getSizeStyle, getTextColor, getVariantStyle } from './utils';

describe('Button', () => {
  it('Button pressed', () => {
    const onPressMock = jest.fn();
    const testIdMock = 'test-id-mock';

    const result = render(
      <Button onPress={onPressMock} testID={testIdMock}>
        Clique me
      </Button>,
    );

    const button = result.getByTestId(testIdMock);

    fireEvent.press(button);

    expect(onPressMock).toBeCalled();
  });

  it('Button disabled pressed', () => {
    const onPressMock = jest.fn();
    const testIdMock = 'test-id-mock';

    const result = render(
      <Button onPress={onPressMock} disabled testID={testIdMock}>
        Clique me
      </Button>,
    );

    const button = result.getByTestId(testIdMock);

    fireEvent.press(button);

    expect(onPressMock).not.toBeCalled();
  });

  it('Button loading renders correctly', () => {
    const onPress = jest.fn();
    const testRenderer = render(
      <Button onPress={onPress} loading>
        Clique me
      </Button>,
    );

    expect(testRenderer.UNSAFE_getByType(Loading)).toBeDefined();
  });

  it('Button loading pressed', () => {
    const onPressMock = jest.fn();
    const testIdMock = 'test-id-mock';

    const result = render(
      <Button onPress={onPressMock} loading testID={testIdMock}>
        Clique me
      </Button>,
    );

    const button = result.getByTestId(testIdMock);

    fireEvent.press(button);

    expect(onPressMock).not.toBeCalled();
  });

  it('Should render Icon component', () => {
    const textToRender = '';

    const result = render(
      <Button
        disabled
        onPress={jest.fn()}
        variant="containedPrimary"
        size="large"
        loading={false}
        Icon={Icons.Small.Right}
        block={false}
      >
        {textToRender}
      </Button>,
    );

    const containerComponent = result.getByTestId('container-icon');
    const iconComponent = within(containerComponent).UNSAFE_getByType(Icons.Small.Right);

    expect(iconComponent).toBeTruthy();
    expect(() => {
      result.UNSAFE_getByType(Text);
    }).toThrowError();
  });

  it('Should render Icon component with text in children', () => {
    const textToRender = 'Clique aqui';

    const result = render(
      <Button
        disabled
        onPress={jest.fn()}
        variant="containedPrimary"
        size="large"
        loading={false}
        Icon={Icons.Small.Right}
        block={false}
      >
        {textToRender}
      </Button>,
    );

    const containerComponent = result.getByTestId('container-icon');
    const TextComponent = result.getByTestId('text-with-icon');
    const iconComponent = within(containerComponent).UNSAFE_getByType(Icons.Small.Right);

    expect(iconComponent).toBeTruthy();
    expect(TextComponent).toBeTruthy();
  });

  it('Should text to render component with prop block true', () => {
    const textToRender = '';

    const result = render(
      <Button
        disabled
        onPress={jest.fn()}
        variant="containedPrimary"
        size="large"
        loading={false}
        Icon={Icons.Small.Right}
      >
        {textToRender}
      </Button>,
    );

    const buttonComponent = result.getByTestId('container-button');

    const viewStyled = StyleSheet.flatten(buttonComponent.props.style);

    expect(viewStyled).not.toHaveProperty('flexDirection', 'row');
  });

  it('Should text to render component with prop block false', () => {
    const textToRender = '';

    const result = render(
      <Button
        disabled
        onPress={jest.fn()}
        variant="containedPrimary"
        size="large"
        loading={false}
        Icon={Icons.Small.Right}
        block={false}
      >
        {textToRender}
      </Button>,
    );

    const buttonComponent = result.getByTestId('container-button');

    const viewStyled = StyleSheet.flatten(buttonComponent.props.style);

    expect(viewStyled).toHaveProperty('flexDirection', 'row');
  });
});

describe('Button snapshot', () => {
  it('default', () => {
    const tree = render(<Button onPress={() => null}>Clique me</Button>).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Button small', () => {
    const tree = render(
      <Button onPress={() => null} size="small">
        Clique me
      </Button>,
    )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Button medium', () => {
    const tree = render(
      <Button onPress={() => null} size="medium">
        Clique me
      </Button>,
    )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Button large', () => {
    const tree = render(
      <Button onPress={() => null} size="large">
        Clique me
      </Button>,
    )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('Utils', () => {
  it('getVariantStyle containedPrimary not disabled', () => {
    const variantStyleMock: ViewStyle = {
      backgroundColor: DefaultTheme.colors.primaryMain,
      borderWidth: DefaultTheme.borders.width.none,
    };

    const variantStyle = getVariantStyle(DefaultTheme, 'containedPrimary', false);

    expect(variantStyle).toEqual(variantStyleMock);
  });

  it('getVariantStyle containedPrimary disabled', () => {
    const variantStyleMock: ViewStyle = {
      backgroundColor: DefaultTheme.colors.primary100,
      borderWidth: DefaultTheme.borders.width.none,
    };

    const variantStyle = getVariantStyle(DefaultTheme, 'containedPrimary', true);

    expect(variantStyle).toEqual(variantStyleMock);
  });

  it('getVariantStyle containedSecondary not disabled', () => {
    const variantStyleMock: ViewStyle = {
      backgroundColor: DefaultTheme.colors.primary200,
      borderWidth: DefaultTheme.borders.width.none,
    };

    const variantStyle = getVariantStyle(DefaultTheme, 'containedSecondary', false);

    expect(variantStyle).toEqual(variantStyleMock);
  });

  it('getVariantStyle containedSecondary disabled', () => {
    const variantStyleMock: ViewStyle = {
      backgroundColor: DefaultTheme.colors.primary100,
      borderWidth: DefaultTheme.borders.width.none,
    };

    const variantStyle = getVariantStyle(DefaultTheme, 'containedSecondary', true);

    expect(variantStyle).toEqual(variantStyleMock);
  });

  it('getVariantStyle outlinedPrimary not disabled', () => {
    const variantStyleMock: ViewStyle = {
      backgroundColor: 'transparent',
      borderWidth: DefaultTheme.borders.width.thin,
      borderColor: DefaultTheme.colors.primary600,
    };

    const variantStyle = getVariantStyle(DefaultTheme, 'outlinedPrimary', false);

    expect(variantStyle).toEqual(variantStyleMock);
  });

  it('getVariantStyle outlinedPrimary disabled', () => {
    const variantStyleMock: ViewStyle = {
      backgroundColor: 'transparent',
      borderWidth: DefaultTheme.borders.width.thin,
      borderColor: DefaultTheme.colors.primary100,
    };

    const variantStyle = getVariantStyle(DefaultTheme, 'outlinedPrimary', true);

    expect(variantStyle).toEqual(variantStyleMock);
  });

  it('getVariantStyle outlinedSecondary not disabled', () => {
    const variantStyleMock: ViewStyle = {
      backgroundColor: 'transparent',
      borderWidth: DefaultTheme.borders.width.thin,
      borderColor: DefaultTheme.colors.neutralGray300,
    };

    const variantStyle = getVariantStyle(DefaultTheme, 'outlinedSecondary', false);

    expect(variantStyle).toEqual(variantStyleMock);
  });

  it('getVariantStyle outlinedSecondary disabled', () => {
    const variantStyleMock: ViewStyle = {
      backgroundColor: 'transparent',
      borderWidth: DefaultTheme.borders.width.thin,
      borderColor: DefaultTheme.colors.neutralGray100,
    };

    const variantStyle = getVariantStyle(DefaultTheme, 'outlinedSecondary', true);

    expect(variantStyle).toEqual(variantStyleMock);
  });

  it('getVariantStyle text not disabled', () => {
    const variantStyleMock: ViewStyle = {
      backgroundColor: 'transparent',
      borderWidth: DefaultTheme.borders.width.none,
    };

    const variantStyle = getVariantStyle(DefaultTheme, 'text', false);

    expect(variantStyle).toEqual(variantStyleMock);
  });

  it('getVariantStyle text disabled', () => {
    const variantStyleMock: ViewStyle = {
      backgroundColor: DefaultTheme.colors.neutralGray100,
      borderWidth: DefaultTheme.borders.width.none,
    };

    const variantStyle = getVariantStyle(DefaultTheme, 'text', true);

    expect(variantStyle).toEqual(variantStyleMock);
  });

  it('getTextColor containedSecondary not disabled', () => {
    const textColor = getTextColor('containedSecondary', false);

    expect(textColor).toEqual('primaryMain');
  });

  it('getTextColor containedSecondary disabled', () => {
    const textColor = getTextColor('containedSecondary', true);

    expect(textColor).toEqual('primary300');
  });

  it('getTextColor outlinedPrimary not disabled', () => {
    const textColor = getTextColor('outlinedPrimary', false);

    expect(textColor).toEqual('primaryMain');
  });

  it('getTextColor outlinedPrimary disabled', () => {
    const textColor = getTextColor('outlinedPrimary', true);

    expect(textColor).toEqual('primary300');
  });

  it('getTextColor outlinedSecondary not disabled', () => {
    const textColor = getTextColor('outlinedSecondary', false);

    expect(textColor).toEqual('neutralGray600');
  });

  it('getTextColor outlinedSecondary disabled', () => {
    const textColor = getTextColor('outlinedSecondary', true);

    expect(textColor).toEqual('neutralGray200');
  });

  it('getTextColor text not disabled', () => {
    const textColor = getTextColor('text', false);

    expect(textColor).toEqual('neutralGray500');
  });

  it('getTextColor text disabled', () => {
    const textColor = getTextColor('text', true);

    expect(textColor).toEqual('neutralGray400');
  });

  it('getTextColor containedPrimary not disabled', () => {
    const textColor = getTextColor('containedPrimary', false);

    expect(textColor).toEqual('neutralWhite');
  });

  it('getTextColor containedPrimary disabled', () => {
    const textColor = getTextColor('containedPrimary', true);

    expect(textColor).toEqual('primary300');
  });

  it('getSizeStyle large', () => {
    const sizeStyleMock: ViewStyle = {
      height: 64,
      minWidth: 64,
      paddingHorizontal: 32,
    };

    const sizeStyle = getSizeStyle(DefaultTheme, 'large', true);

    expect(sizeStyle).toEqual(sizeStyleMock);

    const sizeStyleNoText = getSizeStyle(DefaultTheme, 'large', false);

    expect(sizeStyleNoText.paddingHorizontal).toEqual(undefined);
  });

  it('getSizeStyle small', () => {
    const sizeStyleMock: ViewStyle = {
      height: 30,
      minWidth: 30,
      paddingHorizontal: 16,
    };

    const sizeStyle = getSizeStyle(DefaultTheme, 'small', true);

    expect(sizeStyle).toEqual(sizeStyleMock);

    const sizeStyleNoText = getSizeStyle(DefaultTheme, 'small', false);

    expect(sizeStyleNoText.paddingHorizontal).toEqual(undefined);
  });

  it('getSizeStyle medium', () => {
    const sizeStyleMock: ViewStyle = {
      height: 48,
      minWidth: 48,
      paddingHorizontal: 24,
    };

    const sizeStyle = getSizeStyle(DefaultTheme, 'medium', true);

    expect(sizeStyle).toEqual(sizeStyleMock);

    const sizeStyleNoText = getSizeStyle(DefaultTheme, 'medium', false);

    expect(sizeStyleNoText.paddingHorizontal).toEqual(undefined);
  });
});
