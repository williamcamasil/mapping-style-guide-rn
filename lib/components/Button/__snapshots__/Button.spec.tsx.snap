// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot Button large 1`] = `
<View
  style={{}}
  testID="container-button"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#E95814",
        "borderRadius": 12,
        "borderWidth": 0,
        "display": "flex",
        "height": 64,
        "justifyContent": "center",
        "minWidth": 64,
        "opacity": 1,
        "paddingHorizontal": 32,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#FFFFFF",
          "flexShrink": 1,
          "fontFamily": "Averta CY",
          "fontSize": 16,
          "fontWeight": "bold",
          "letterSpacing": -0.02,
          "lineHeight": 19,
        }
      }
      testID="text-with-icon"
    >
      Clique me
    </Text>
  </View>
</View>
`;

exports[`Button snapshot Button medium 1`] = `
<View
  style={{}}
  testID="container-button"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#E95814",
        "borderRadius": 12,
        "borderWidth": 0,
        "display": "flex",
        "height": 48,
        "justifyContent": "center",
        "minWidth": 48,
        "opacity": 1,
        "paddingHorizontal": 24,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#FFFFFF",
          "flexShrink": 1,
          "fontFamily": "Averta CY",
          "fontSize": 16,
          "fontWeight": "bold",
          "letterSpacing": -0.02,
          "lineHeight": 19,
        }
      }
      testID="text-with-icon"
    >
      Clique me
    </Text>
  </View>
</View>
`;

exports[`Button snapshot Button small 1`] = `
<View
  style={{}}
  testID="container-button"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#E95814",
        "borderRadius": 12,
        "borderWidth": 0,
        "display": "flex",
        "height": 30,
        "justifyContent": "center",
        "minWidth": 30,
        "opacity": 1,
        "paddingHorizontal": 16,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#FFFFFF",
          "flexShrink": 1,
          "fontFamily": "Averta CY",
          "fontSize": 14,
          "fontWeight": "bold",
          "letterSpacing": -0.02,
          "lineHeight": 17,
        }
      }
      testID="text-with-icon"
    >
      Clique me
    </Text>
  </View>
</View>
`;

exports[`Button snapshot default 1`] = `
<View
  style={{}}
  testID="container-button"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#E95814",
        "borderRadius": 12,
        "borderWidth": 0,
        "display": "flex",
        "height": 48,
        "justifyContent": "center",
        "minWidth": 48,
        "opacity": 1,
        "paddingHorizontal": 24,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#FFFFFF",
          "flexShrink": 1,
          "fontFamily": "Averta CY",
          "fontSize": 16,
          "fontWeight": "bold",
          "letterSpacing": -0.02,
          "lineHeight": 19,
        }
      }
      testID="text-with-icon"
    >
      Clique me
    </Text>
  </View>
</View>
`;
